### PR TITLE
OCP4 allowed registries

### PR DESCRIPTION
**Pull request by shaneboulden**
#### Description:

Checks the cluster-wide OCP image configuration to ensure that at least one `allowedRegistriesForImport` is specified. Based on the existing [ocp_idp_no_htpasswd](https://github.com/ComplianceAsCode/content/tree/master/applications/openshift/ocp_idp_no_htpasswd) check.

#### Rationale:

The configuration `allowedRegistriesForImport` limits the container image registries from which normal users may import images. It only affects the image API (so for example `oc import-image`, ImageStreams, etc).

This important to control because a user who can stand up a malicious registry can then import content from it which claims to include the SHAs of legitimate content layers.

### CODE REVIEW

**Comment by mildas** [comment]
**Changes identified:**
Rule ocp_allowed_registries_for_import:
 The rule doesn't occur in any profile nor product.
 OVAL check is newly added.

File: applications/openshift/registry/ocp_allowed_registries_for_import/oval/shared.xml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	    <unix:object object_ref="object_file_for_{{{ rule_id }}}"/>
+    </unix:file_test>
+
+    <unix:file_object id="object_file_for_{{{ rule_id }}}" version="1">
+      <unix:filepath var_ref="{{{ rule_id }}}_dump_location"/>
+    </unix:file_object>
+
+    <ind:yamlfilecontent_object id="object_{{{ rule_id }}}" version="1">
+      <ind:filepath var_ref="{{{ rule_id }}}_dump_location"/>
+      <ind:yamlpath>.spec.allowedRegistriesForImport[:]</ind:yamlpath>
```
Comment (shaneboulden) [inline applications/openshift/registry/ocp_allowed_registries_for_import/oval/shared.xml:38]: @matejak is this sufficient to test that at least one `allowedRegistriesForImport` exists?

File: applications/openshift/registry/ocp_allowed_registries_for_import/rule.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    image registries from which normal users may import images. This is important
+    to control, as a user who can stand up a malicious registry can then import
+    content which claims to include the SHAs of legimitate content layers.
+
+rationale: |-
+    Allowed registries for import should be specified to limit the registries 
+    from which users may import images.
+
+identifiers:
+    cce@ocp4: 84210-1
```
Comment (shaneboulden) [inline applications/openshift/registry/ocp_allowed_registries_for_import/rule.yml:16]: Placeholder for now

**Comment by JAORMX** [comment]
Could you rebase? we just added e2e tests for ocp4 changes. If this would be added to the moderate profile it would get tested

File: applications/openshift/registry/ocp_allowed_registries_for_import/rule.yml (hunk 2 above)
Comment (JAORMX) [reply applications/openshift/registry/ocp_allowed_registries_for_import/rule.yml:16]: Let's remove this for now, since this is making the build break

**Comment by mildas** [comment]
**Changes identified:**
Rule ocp_allowed_registries_for_import:
 OVAL check is newly added.
 The rule doesn't occur in any profile nor product.

File: applications/openshift/registry/ocp_allowed_registries_for_import/rule.yml (hunk 2 above)
Comment (shaneboulden) [reply applications/openshift/registry/ocp_allowed_registries_for_import/rule.yml]: removed and rebased

File: applications/openshift/registry/ocp_allowed_registries_for_import/oval/shared.xml (hunk 1 above)
Comment (shaneboulden) [reply applications/openshift/registry/ocp_allowed_registries_for_import/oval/shared.xml:38]: @evgenyz does this look sufficient to test that at least one `allowedRegistriesForImport` entry exists? Or is there another test required?

Comment (evgenyz) [reply applications/openshift/registry/ocp_allowed_registries_for_import/oval/shared.xml:38]: Yes, that's correct

**Comment by mildas** [comment]
**Changes identified:**
Rule ocp_allowed_registries_for_import:
 OVAL check is newly added.
Profile e8 on ocp4:
 Rule ocp_allowed_registries_for_import added to e8 profile.
Profile moderate on ocp4:
 Rule ocp_allowed_registries_for_import added to moderate profile.

**Recommended tests to execute:**
 build_product ocp4
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ocp4-ds.xml ocp_allowed_registries_for_import
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-ocp4-ds.xml e8
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-ocp4-ds.xml moderate

**Comment by JAORMX** [comment]
@evgenyz seems it's failing with:

```
 Traceback (most recent call last):
  File "/content/build-scripts/yaml_to_shorthand.py", line 95, in <module>
    main()
  File "/content/build-scripts/yaml_to_shorthand.py", line 82, in main
    p.validate_rules(loader.all_rules, loader.all_groups)
  File "/content/ssg/build_yaml.py", line 307, in validate_rules
    raise ValueError(msg)
ValueError: Rule 'ocp_allowed_registries_for_import' was not found in the benchmark. Please remove rule 'ocp_allowed_registries_for_import' from profile 'e8' before proceeding.
```

any idea what's missing?

**Comment by yuumasato** [comment]
> @evgenyz seems it's failing with:
> 
> ```
>  Traceback (most recent call last):
>   File "/content/build-scripts/yaml_to_shorthand.py", line 95, in <module>
>     main()
>   File "/content/build-scripts/yaml_to_shorthand.py", line 82, in main
>     p.validate_rules(loader.all_rules, loader.all_groups)
>   File "/content/ssg/build_yaml.py", line 307, in validate_rules
>     raise ValueError(msg)
> ValueError: Rule 'ocp_allowed_registries_for_import' was not found in the benchmark. Please remove rule 'ocp_allowed_registries_for_import' from profile 'e8' before proceeding.
> ```
> 
> any idea what's missing?

@JAORMX @shaneboulden `applications/openshift/registry` needs a `group.yml` file.
You can get inspiration from similar files in other directories.

**Comment by mildas** [comment]
**Changes identified:**
Rule ocp_allowed_registries_for_import:
 OVAL check is newly added.
Profile e8 on ocp4:
 Rule ocp_allowed_registries_for_import added to e8 profile.
Profile moderate on ocp4:
 Rule ocp_allowed_registries_for_import added to moderate profile.

**Recommended tests to execute:**
 build_product ocp4
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ocp4-ds.xml ocp_allowed_registries_for_import
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-ocp4-ds.xml e8
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-ocp4-ds.xml moderate

**Comment by JAORMX** [comment]
LGTM

**Pull request merged**